content/06.model-complexity.md

```diff
@@ -1,34 +1,39 @@
-For example, among various regularization methods (ridge regression, LASSO regression, and elastic-net)[@doi:10.1111/j.1467-9868.2005.00503.x], hybrid applications of LASSO for capturing combinations of rare variants have proven beneficial [@doi:10.1016/j.ajhg.2008.06.024; @doi:10.1186/1753-6561-5-s9-s113] .
-In the context of rare immune cell signature discovery, elastic-net regression was found to outperform other regression approaches [@doi:10.1111/j.1467-9868.2005.00503.x; @doi:10.1016/j.compbiomed.2015.10.008; @doi:10.1186/1471-2105-14-198; @doi:10.1186/s12859-019-2994-z]. 
-Thus regularization methods like LASSO or elastic-net have been methods of choice while working with rare observations; use of these regularization approaches should be considered while working with rare disease datasets. 
+Recent studies suggest that there are limitations to decision tree-based ensemble methods when applied to rare disease datasets, leading to adoption of cascade learning, a variant of ensemble learning [@doi:10.1007/s11634-019-00354-x; @pmid:30815073].
+In cascade learning, multiple methods leveraging distinct underlying assumptions are used in tandem to capture stable patterns existing in the dataset [@doi:10.1109/cvpr.2001.990537; @doi:10.1007/978-3-540-75175-5_16; @doi:10.1109/icpr.2004.1334680]. 
+For example, a cascade learning approach for identifying rare disease patients from electronic health record data utilized independent steps for feature extraction (word2vec [@arxiv:1301.3781]), followed by preliminary prediction with ensembled decision trees, and then prediction refinement using data similarity metrics [@pmid:30815073]. 
+Combining these three unrelated methods resulted in better overall prediction when implemented on a silver standard dataset, than other methods applied in isolation.
+In addition to cascade learning, other approaches that better represent rare classes like inverse class weighting, oversampling [doi:10.1613/jair.953] may also improve ML models that use rare disease data. 
 
+The presence of label-noise and sparsity in the data can also lead to overfitting of models to the training data, meaning that the models show high prediction accuracy on the training data but low prediction accuracy (and large prediction errors) on new evaluation data. 
+Overfit models tend to rely on variables that are unique to the training data (for example, the calibration of the instrument that was used to generate the training data), and not generalizable to new data (e.g., data generated on the same instrument that has been recalibrated). [@doi:10.1073/pnas.1900654116]
+In such cases, regularization can not only protect ML models from poor generalizability caused by overfitting, but also reduce model complexity by reducing the feature space available for training. (Figure[@fig:2]C)
+Regularization is a collection of methods (generally regression methods) that can help reduce prediction errors by various approaches. 
```

Comment:
I don't know that I agree that "regularization is a collection of methods (generally regression methods)" vs. it's the approach/concept of adding a penalty to increase generalizability or some other desirable property and control overfitting. So if we take a VAE as an example, use of KL divergence in the loss function is often considered/talked about as regularization.

Reply:
I think this comment could be addressed by updating this sentence to be more general. I think using the regression examples is fine. But then maybe 1-2 sentences that talk about other scenarios at the end of the paragraph...

Reply:
Fair point..

Reply:
```suggestion
Regularization is an approach by which a penalty is added to the model when large prediction errors occur. 
```